plugin/plugin_manager.go

```diff
@@ -0,0 +1,188 @@
+package plugin
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	getter "github.com/hashicorp/go-getter"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/odpf/optimus/config"
+	"github.com/odpf/optimus/plugin/yaml"
+)
+
+var (
+	// for plugin installation, discovery & sync on both client and server
+	PluginsDir         = ".plugins"
+	PluginsArchiveName = "yaml-plugins.zip"
+)
+
+type IPluginManager interface {
+	Install(dst string, sources ...string) error
+	Archive(name string) error
+	UnArchive(src, dest string) error
+}
+
+func NewPluginManager() *PluginManager {
+	pluginLoggerOpt := &hclog.LoggerOptions{
+		Name:   "plugin-manager",
+		Output: os.Stdout,
+		Level:  hclog.Info,
+	}
+	logger := hclog.New(pluginLoggerOpt)
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		logger.Error(fmt.Sprintf("Error getting pwd: %s", err))
+	}
+	client := &getter.Client{
+		Src:  "",
+		Dst:  "",
+		Pwd:  pwd,
+		Mode: getter.ClientModeAny,
+	}
+	return &PluginManager{
+		logger: logger,
+		client: client,
+	}
+}
+
+type PluginManager struct {
+	logger hclog.Logger
+	client *getter.Client
+}
+
+func (p *PluginManager) installOne(dst, src string) error {
+	p.client.Src = src
+	p.client.Dst = dst
+
+	if err := p.client.Get(); err != nil {
+		p.logger.Error(fmt.Sprintf("Error installing plugin from [%s]", src))
+		return err
+	}
+	p.logger.Info(fmt.Sprintf("Success installing plugin from [%s]", src))
+	return nil
+}
+
+func (p *PluginManager) Install(dst string, sources ...string) error {
+	for _, src := range sources {
+		if err := p.installOne(dst, src); err != nil {
+			p.logger.Error("*** Plugin Installation Aborted !!. Please check if plugin.artifacts are correct")
+			return err
+		}
+	}
+	p.logger.Info(fmt.Sprintf("Success installing plugin in dir=%s", dst))
+	return nil
+}
+
+func SanitizeArchivePath(d, t string) (v string, err error) {
```

Comment:
Why are we using this function outside this module ? if not it should start with lowercase letter